### PR TITLE
Bring in huge upstream update to ace android

### PR DESCRIPTION
**Pull request by shareefalis**

### CODE REVIEW

**Comment by shareefalis** [comment]
Need to add ccache to build.. Will work on that for monday

File: .travis.yml [hunk 1]
```diff
@@ -92,8 +92,7 @@ before_script:
 - android list target -c
 script:
 - ./Tools/git_version.sh
-- ./Tools/travis_script.sh
```
Comment (shareefalis) [inline .travis.yml:93]: TODO: After ccache, need to revert the changes there.

File: Tools/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -xe
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $DIR/..
+
+python prepare.py -C || true
+if [ -d build ]; then rm -fr build; fi
+python prepare.py  -DENABLE_WEBRTC_AEC=ON -DENABLE_H263=YES -DENABLE_FFMPEG=YES -DENABLE_NON_FREE_CODECS=ON -DENABLE_GPL_THIRD_PARTIES=ON -DENABLE_AMRWB=NO -DENABLE_AMRNB=NO -DENABLE_OPENH264=YES -DENABLE_X264=NO -DENABLE_G729=NO -DENABLE_MPEG4=NO -DENABLE_H263P=NO -DENABLE_ILBC=NO -DENABLE_ISAC=NO -DENABLE_SILK=NO -DENABLE_VCARD=ON -DLINPHONE_BUILDER_LATEST=YES -DENABLE_RELATIVE_PREFIX=YES -DENABLE_GSM=YES -DENABLE_OPUS=YES -DENABLE_SILK=YES -DENABLE_BV16=NO -DENABLE_MKV=YES -DENABLE_SPEEX=YES -DLINPHONE_BUILDER_LATEST=NO
```
Comment (shareefalis) [inline Tools/prepare.sh]: TODO: add: "-DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"

**Comment by patrickmwatson** [comment]
+1 LGTM
